### PR TITLE
Increase code coverage for DeviceLogic

### PR DESCRIPTION
**Pull request by handeadiguzel**

### CODE REVIEW

File: DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs [hunk 1]
```diff
@@ -384,29 +381,15 @@ await _virtualDeviceStorage.AddOrUpdateDeviceAsync(new InitialDeviceConfig()
             {
                 throw new DeviceRequiredPropertyNotFoundException("Required DeviceProperties not found");
             }
-
-            if ((dynamicMetaObjectProvider = device.DeviceProperties as IDynamicMetaObjectProvider) != null)
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs]: Can you check if RM works okay ?

File: DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs [hunk 2]
```diff
@@ -30,36 +31,39 @@ public class DeviceLogicTests
...
-            _deviceRegistryListRepositoryMock = new Mock<IDeviceRegistryListRepository>();
-            _virtualDeviceStorageMock = new Mock<IVirtualDeviceStorage>();
-            _configProviderMock = new Mock<IConfigurationProvider>();
-            _securityKeyGeneratorMock = new Mock<ISecurityKeyGenerator>();
-            _deviceRulesLogicMock = new Mock<IDeviceRulesLogic>();
-            _deviceLogic = new DeviceLogic(_iotHubRepositoryMock.Object, _deviceRegistryCrudRepositoryMock.Object,
-                _deviceRegistryListRepositoryMock.Object, _virtualDeviceStorageMock.Object,
-                _securityKeyGeneratorMock.Object, _configProviderMock.Object, _deviceRulesLogicMock.Object);
-            fixture = new Fixture();
+            this._iotHubRepositoryMock = new Mock<IIotHubRepository>();
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs]: I think if you run resharper code cleanup, it actually removes this at obvious places.

File: DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs [hunk 3]
```diff
@@ -282,22 +289,85 @@ public void ExtractLocationsDataTest()
...
+        {
+            var deviceId = this.fixture.Create<string>();
+            var isEnabled = this.fixture.Create<bool>();
+            var device = this.fixture.Create<DeviceModel>();
+            this._iotHubRepositoryMock.Setup(mock => mock.UpdateDeviceEnabledStatusAsync(deviceId, isEnabled)).ReturnsAsync(new Device());
+            this._deviceRegistryCrudRepositoryMock.Setup(mock => mock.UpdateDeviceEnabledStatusAsync(deviceId, isEnabled)).ReturnsAsync(device);
+            var res = await this._deviceLogic.UpdateDeviceEnabledStatusAsync(deviceId, isEnabled);
+            Assert.Equal(res, device);
+
+            this._deviceRegistryCrudRepositoryMock.Setup(mock => mock.UpdateDeviceEnabledStatusAsync(deviceId, isEnabled)).Throws<Exception>();
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs]: you can use SetupSequence instead of Setup multiple times

Comment (Inspiravetion) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs]: example?

File: DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs (hunk 1 above)
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure/BusinessLogic/DeviceLogic.cs]: yeap it's running fine. The code that I've deleted was hit when device was dynamic...

File: DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs (hunk 2 above)
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs]: this is actually the result of code cleanup.. maybe our resharper's are configured different.

File: DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs (hunk 3 above)
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs]: yeah I saw that in your PRs.. it's cool. But let's use it in shorter tests.

File: DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs [hunk 4]
```diff
@@ -282,22 +289,86 @@ public void ExtractLocationsDataTest()
...
+            var isEnabled = this.fixture.Create<bool>();
+            var device = this.fixture.Create<DeviceModel>();
+            this._iotHubRepositoryMock.Setup(mock => mock.UpdateDeviceEnabledStatusAsync(deviceId, isEnabled)).ReturnsAsync(new Device());
+            this._deviceRegistryCrudRepositoryMock.SetupSequence(mock => mock.UpdateDeviceEnabledStatusAsync(deviceId, isEnabled))
+                .ReturnsAsync(device)
+                .Throws<Exception>();
+            var res = await this._deviceLogic.UpdateDeviceEnabledStatusAsync(deviceId, isEnabled);
+            Assert.Equal(res, device);
+
+            this._iotHubRepositoryMock.Setup(mock => mock.UpdateDeviceEnabledStatusAsync(deviceId, !isEnabled)).ReturnsAsync(new Device());
```
Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs]: you can use setup sequence here also

Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs]: the input is different: !isEnabled

**Pull request merged**